src/System.IO.FileSystem/tests/Enumeration/ExampleTests.netcoreapp.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO.Enumeration;
+using Xunit;
+
+namespace System.IO.Tests.Enumeration
+{
+    // For tests that cover examples from documentation, blog posts, etc.
+    public class ExampleTests : FileSystemTest
```

Comment:
Are these cases not covered elsewhere?

Reply:
I hear what you're saying, but I disagree with the assessment of usefulness. In general I've been trying to add "scenario" tests that match our docs (notably in Configuration and IsolatedStorage). It would be particularly embarrassing to break examples we've given. They are essentially end-to-end (e2e) tests if you want to think of it that way.

Reply:
How can this possibly be an end to end test? It doesn't even enumerate anything.

This test is no different in its unit/functional test-ness from any of the thousands of other tests we have, and it should be named and organized in a way that highlights what it's verifying.

In a year if I came across this test I would simply delete it, as it's entirely unclear what unique value it's providing (or, rather, it appears to me to be providing no unique value).

Reply:
Ugh, I apparently squashed incorrectly. There _was_ an assert.

Reply:
Assert added back. I believe _strongly_ in having these. My experience has shown having tests like these, even if they intersect with "normal" tests to be useful. Yes, this first one is simple- I'm writing others that aren't quite as simple. I stopped in the middle when I realized the other tests were still disabled.

Reply:
I'm still not a fan, even with an assert. What is this testing that https://github.com/JeremyKuhne/corefx/blob/22e9e644100bd4de8635e40d2c4776c72b2be776/src/System.IO.FileSystem/tests/Enumeration/IncludePredicateTests.netcoreapp.cs isn't? If it's testing something distinct, then the test name should convey that, and it should be grouped with other like tests. If it's not testing something distinct, then it's not adding value.

You say it's matching our docs? What docs? There's no link here that connects it to that. How does someone maintaining these tests besides you understand the purpose? Or how do you in a year?

I appreciate your good here, and I would actually like to see a solution that let's us have docs pull examples directly from the repo, live, such that we know the exact code being shown in a particular place in the docs is correct. But this isn't that.

Reply:
The link point is relevant, I'll add it when I add the others.

I agree that it would be awesome (in theory) if docs could pull directly from the repo. They don't have test infrastructure, etc.